### PR TITLE
Use QDateTime::currentDateTimeUtc instead of the non utc version

### PR DESCRIPTION
**Pull request by ogoffart**
QDateTime::currentDateTime is terribly slow.

### CODE REVIEW

**Comment by jturcotte** [comment]
Could you explain why they are equivalent? Won't this offset all timestamps vs the database?

**Comment by ogoffart** [comment]
The difference between the two is the Qt::TimeSpec.
But any computation on the QDateTime will result in the same results. In particular toMSecsSinceEpoch.

File: src/libsync/propagateupload.cpp [hunk 1]
```diff
@@ -53,7 +53,7 @@ Q_LOGGING_CATEGORY(lcPropagateUpload, "sync.propagator.upload", QtInfoMsg)
 static bool fileIsStillChanging(const SyncFileItem &item)
 {
     const QDateTime modtime = Utility::qDateTimeFromTime_t(item._modtime);
-    const qint64 msSinceMod = modtime.msecsTo(QDateTime::currentDateTime());
+    const qint64 msSinceMod = modtime.msecsTo(QDateTime::currentDateTimeUtc());
```
Comment (guruz) [inline src/libsync/propagateupload.cpp]: What about
```pi._modtime = Utility::qDateTimeFromTime_t(_item->_modtime);```
Is there anything that need to be done?

File: src/libsync/syncresult.cpp [hunk 2]
```diff
@@ -85,7 +85,7 @@ QString SyncResult::statusString() const
 void SyncResult::setStatus(Status stat)
 {
     _status = stat;
-    _syncTime = QDateTime::currentDateTime();
+    _syncTime = QDateTime::currentDateTimeUtc();
```
Comment (jturcotte) [inline src/libsync/syncresult.cpp]: This actually looks unused now.

File: src/libsync/propagatedownload.cpp [hunk 3]
```diff
@@ -673,7 +673,7 @@ namespace { // Anonymous namespace for the recall feature
             dotLocation = recallFileName.size();
         }
 
-        QString timeString = QDateTime::currentDateTime().toString("yyyyMMdd-hhmmss");
+        QString timeString = QDateTime::currentDateTimeUtc().toString("yyyyMMdd-hhmmss");
         recallFileName.insert(dotLocation, "_.sys.admin#recall#-" + timeString);
```
Comment (jturcotte) [inline src/libsync/propagatedownload.cpp]: This one could actually stay, but I guess users shouldn't really see this anyway (nor do I fully understand why we append a date there).

**Pull request merged**